extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +151,235 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(win, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+        'Cross-domain frame parameters missing ${this.type_}');
+    const frameData = this.useExistingOrCreateCrossDomainIframe(
+        win, frameUrl, transportOptions['extraData']);
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
```

Comment:
Copying conversation from old PR:

@lannka:
should you use SubscriptionApi at host side?

Me:
Keith, what do you think? It does seem like this would be better on the AMP side. But AFAICT the 3p side would have to revert to window.addEventListener("message", ...) which loses a lot of what we gain from iframeMessagingClient.